### PR TITLE
Automated cherry pick of #116550: Add integration test for DefaultBinder

### PR DESCRIPTION
**Pull request by alculquicondor**
Cherry pick of #116550 on release-1.26.

#116550: Add integration test for DefaultBinder

For details on the cherry pick process, see the [cherry pick requests](https://git.k8s.io/community/contributors/devel/sig-release/cherry-picks.md) page.

```release-note
Fixes a 1.25 regression in the pod binding subresource to honor the `metadata.uid` precondition.
This allows kube-scheduler to ensure it is assigns node names to the same instances of pods it made scheduling decisions for.
```

### CODE REVIEW

File: pkg/features/kube_features.go [hunk 1]
```diff
@@ -662,6 +662,13 @@ const (
 	// Requires the CRI implementation supports supplying the required stats.
 	PodAndContainerStatsFromCRI featuregate.Feature = "PodAndContainerStatsFromCRI"
 
+	// owner: @alculquicondor
+	// deprecated: v1.24
+	//
+	// Enables preconditions checks for UID and resourceVersion in the pod/binding
```
Comment (liggitt) [inline pkg/features/kube_features.go:668]: just UID, right? resourceVersion appears to have worked previously

Comment (liggitt) [reply pkg/features/kube_features.go:668]: can you keep the feature-gate commit separate from the picked commits?

**Comment by liggitt** [comment]
Something went wrong: failed to acquire k8s binaries: U=file:///go/src/k8s.io/kubernetes/_output/gcs-stage R=v1.26.3-3+df96302de9bd79 get-kube.sh failed: error during /workspace/get-kube.sh: exit status 1

I think the e2e failure is https://github.com/kubernetes/release/issues/2972

**Comment by liggitt** [comment]
/lgtm
/approve

though the e2e will likely fail until https://github.com/kubernetes/release/issues/2972 is resolved

File: pkg/features/kube_features.go (hunk 1 above)
Comment (liggitt) [reply pkg/features/kube_features.go]: if this is a regression (you linked to https://github.com/kubernetes/kubernetes/pull/110646 which is where this regressed), then I don't think we need the feature gate and can just backport the fix

**Comment by liggitt** [comment]
waiting on https://github.com/kubernetes/kubernetes/pull/116808 to fix CI

**Comment by alculquicondor** [comment]
ping @kubernetes/release-managers

**Review by xmudrii — APPROVED** [review]

**Pull request merged**